ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -0,0 +1,376 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+@objc(MSFBottomSheetControllerDelegate)
+public protocol BottomSheetControllerDelegate: AnyObject {
+    /// Called after the sheet fully expanded.
+    @objc optional func bottomSheetControllerDidExpand(_ controller: BottomSheetController)
+
+    /// Called after the sheet fully collapsed.
+    @objc optional func bottomSheetControllerDidCollapse(_ controller: BottomSheetController)
+}
+
+@objc(MSFBottomSheetController)
+public class BottomSheetController: UIViewController {
+
+    /// Initializes the bottom sheet controller.
+    /// - Parameter contentViewController: The view controller that manages the bottom sheet content.
+    ///
+    /// By default the root view of `contentViewController` will be sized automatically to fill the available area,
+    /// respecting the provided `preferredExpandedHeightFraction` multiplier.
+    /// Alternatively, the content can size itself by setting `respectsPreferredContentSize` to true and providing a `preferredContentSize`.
+    @objc public init(with contentViewController: UIViewController) {
+        self.contentViewController = contentViewController
+        super.init(nibName: nil, bundle: nil)
+
+        addChild(contentViewController)
+        contentViewController.didMove(toParent: self)
+    }
+
+    required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    /// A scroll view in `contentViewController`'s view hierarchy.
+    /// Provide this to ensure the bottom sheet pan gesture recognizer coordinates with the scroll view to enable scrolling based on current bottom sheet position and content offset.
+    @objc open var hostedScrollView: UIScrollView?
+
+    /// Indicates if the bottom sheet is expandable.
+    @objc open var isExpandable: Bool = true {
+        didSet {
+            if isExpandable != oldValue {
+                if isExpandable {
+                    resizingHandleView.isHidden = false
+                    panGestureRecognizer.isEnabled = true
+                } else {
+                    resizingHandleView.isHidden = true
+                    panGestureRecognizer.isEnabled = false
+                }
+                move(to: .collapsed, animated: false)
+            }
+        }
+    }
+
+    /// Indicates if `preferredContentSize` of `contentViewController` should be respected.
+    /// Upper limits on the content size are still enforced, depending on the area available to `BottomSheetController`.
```

Comment:
If we don't respect preferredContentSize, we will size the sheet content ourselves - attempt to expand it as far as it can go.

About the upper limits - if the child's preferred height would make the expanded sheet taller than the space available to us, we'll clamp the height.